src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -51,7 +51,11 @@ public function __construct(ContainerInterface $container, $resource, array $opt
     public function getRouteCollection()
     {
         if (null === $this->collection) {
+            $this->collection = new RouteCollection();
```

Comment:
Otherwise you could do this logic in basically every method that throws an exception.